scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -51,4 +65,51 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
             analyzer.analyze();
         };
     }
+
+    List<Issue> runExternalAnalyzers(Project project) {
+        StringBuilder newImports = new StringBuilder();
+        StringBuilder tomlDependencies = new StringBuilder();
+        int importCounter = 0;
```

Comment:
Yes, since the size of the import counter is the same as analyzer count will make this change.

This value is used to correct the start and end lines of issues found on the in-memory [import-generated Ballerina file](https://github.com/Xenowa/ballerina-scan-tool/blob/main/ScanCommand/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java#L453).

Reply:
After the change, a spot bugs warning is generated as the variable is unused. Will be adding this implementation in the next PR.

Reply:
Since the generation of imports is now done via a generated in-memory Ballerina file, having this functionality is no longer required.